### PR TITLE
feat: transient mode

### LINKED ISSUE #623: Open transient sidekick if project not configured
Today, when browsing on a URL whose project hasn't been configured in your sidekick yet, you can click the extension icon and select "add this project" to persist it and open the sidekick on affiliated URLs.

It would be helpful if the extension could guess the project config based on the URL and open the sidekick in a transient mode, allowing the user to use it without having to add it to the config first.

### PR DESCRIPTION
**Pull request by rofe**
Fix #623

### CODE REVIEW

File: src/extension/module.js [hunk 1]
```diff
@@ -1135,6 +1136,26 @@ import sampleRUM from './rum.js';
     sk.switchEnv('preview');
   }
 
+  /**
+   * Adds the transient plugin to the sidekick.
+   * @private
+   * @param {Sidekick} sk The sidekick
+   */
+  function addTransientPlugin(sk) {
```
Comment (dylandepass) [inline src/extension/module.js:1144]: Maybe `addTransientAddPlugin` might be a better name? Just a thought

Comment (rofe) [reply src/extension/module.js:1144]: I've been mulling over this function name... had `addAddProjectPlugin` at first 😂 

How about `addTransientAddProjectPlugin`?

File: src/extension/utils.js [hunk 2]
```diff
@@ -729,3 +740,85 @@ export function toggleDisplay(cb) {
...
+    }
+  } catch (e) {
+    log.error(`updateAdminAuthHeaderRules: ${e.message}`);
+  }
+}
+
+export async function storeAuthToken(owner, repo, token, exp) {
+  const handle = `${owner}/${repo}`;
+  const projects = await getConfig('session', 'hlxSidekickProjects') || [];
+  const i = projects.indexOf(handle);
```
[the remark this answers is not in the document]
Comment (rofe) [reply src/extension/utils.js:800]: Caught me being lazy 😬  sometimes more is better

**Pull request merged**